apps/google-analytics-4/frontend/src/components/config-screen/assign-content-type/ContentTypeWarning.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect } from 'react';
+import { Box, Tooltip } from '@contentful/f36-components';
+import { WarningIcon, ErrorCircleIcon } from '@contentful/f36-icons';
+import { styles } from 'components/config-screen/assign-content-type/AssignContentType.styles';
+import {
+  NO_CONTENT_TYPE_ERR_MSG,
+  NO_SLUG_WARNING_MSG,
+  REMOVED_FROM_SIDEBAR_WARNING_MSG,
+  getContentTypeDeletedMsg,
+  getSlugFieldDeletedMsg,
+  WarningTypes,
+} from './constants/warningMessages';
+import { ContentWarningTypes } from 'types';
+
+interface Props {
+  contentTypeId: string;
+  slugField: string;
+  isSaved: boolean;
+  isInSidebar: boolean;
+  isContentTypeInOptions: boolean;
+  isSlugFieldInOptions: boolean;
+}
+
+const ContentTypeWarning = (props: Props) => {
+  const {
+    contentTypeId,
+    slugField,
+    isSaved,
+    isInSidebar,
+    isContentTypeInOptions,
+    isSlugFieldInOptions,
+  } = props;
+
+  const [warningType, setWarningType] = useState<ContentWarningTypes>(WarningTypes.Empty);
+  const [tooltipContent, setTooltipContent] = useState<string>('');
+
+  useEffect(() => {
+    const getTooltipContent = () => {
+      let content = '';
+
+      // Error states
+      if (!contentTypeId) {
+        setWarningType(WarningTypes.Error);
```

Comment:
We might consider moving this and the one below to the bottom, so that error states take precedence on the warning type. (Imagine you have an error and a warning, i think we want the error icon to show.)

Reply:
Agreed, that makes sense!